### ISSUE TITLE
Project settings hold editor infos

### ISSUE THREAD

**Issue by QbieShay** [godotengine/godot#24446]
**Godot version:**
11d7738

**Issue description:**
Seems like Godot is storing editor specific data inside the project settings.
I believe this is a leftover of some sort before the editor had its own settings?
![image](https://user-images.githubusercontent.com/7917475/50152927-4458e480-02c5-11e9-8220-7842b7949b49.png)

**Bonus:**
Why is there a "Active" toggle that has as a tooltip "Internal Editor Setting,don't touch"?
Should it be directly renamed to "DONT TOUCH THIS PLS" ?

**Comment by akien-mga** [comment]
Editor Settings are shared between all projects, so it can make sense to have some project-specific settings to control editor behaviour in Project Settings (like `Main Run Args`, to define how you want *this* project to be run in the editor).

About `editor/active`, I'm not sure what it's used for. It doesn't seem to be used directly anywhere:
```
$ rg "editor/active"
main/main.cpp
1341:   GLOBAL_DEF("editor/active", editor);

doc/classes/ProjectSettings.xml
365:            <member name="editor/active" type="bool" setter="" getter="">
```
and I couldn't find any composed string that would lead to using this setting either (grepping for `"active"`).

**Comment by akien-mga** [comment]
Doesn't seem to be used much even in `1.0-stable`:
```
$ rg 'editor_active'
main/main.cpp
928:            Globals::get_singleton()->set("editor_active",true);
```

**Comment by QbieShay** [comment]
@akien-mga is project metadata already existing? If so where? I use custom run args for debugging purposes..

**Comment by akien-mga** [comment]
Yes, there's `~/.config/godot/projects/<project folder>/project_metadata.cfg`. You can access it with `EditorSettings::get_singleton()->set_project_metadata` and `EditorSettings::get_singleton()->get_project_metadata`.

**Comment by QbieShay** [comment]
Thanks!

**Comment by Calinou** [comment]
`editor/active` was removed in #33333.

Does it still make sense to have **Reimport Missing Imported Files**? I can see an use case for disabling **Convert Text Scenes To Binary On Export** (for debugging purposes), but not so much for that one.
